--- /dev/null
+++ ../Classes/Service/UrlService.php
@@ -81,6 +81,9 @@
             }
             $url = $site->getRouter()->generateUri($pageId, $queryParts);
             if (!empty($alternativeBaseUrl)) {
+                if (strpos($alternativeBaseUrl, '$') === 0) {
+                    $alternativeBaseUrl = getenv(substr($alternativeBaseUrl, 1));
+                }
                 $alternativeBaseUrl = new Uri($alternativeBaseUrl);
                 $url = $url->withHost($alternativeBaseUrl->getHost());
                 $url = $url->withScheme($alternativeBaseUrl->getScheme());
