--- /dev/null
+++ ../Classes/Indexer/Types/Page.php
@@ -684,7 +684,7 @@
                     // add content elements with sys_language_uid = -1 to all language versions of this page
                     if ($ttContentRow['sys_language_uid'] == -1) {
                         foreach ($this->sysLanguages as $sysLang) {
-                            if ($sysLang['uid'] != -1) {
+                            if ($sysLang['uid'] != -1 && isset($pageContent[$sysLang['uid']])) {
                                 $pageContent[$sysLang['uid']] .= $content;
                             }
                         }
@@ -1067,7 +1067,7 @@
             $processedData = $this->filesProcessor->process($this->cObj, [], $configuration, $processedData);
             $fileReferenceObjects = array_merge($fileReferenceObjects, $processedData['files']);
         }
-        
+
         return $fileReferenceObjects;
     }
 
