--- /dev/null
+++ ../Classes/Backend/LoginProvider/Oauth2LoginProvider.php
@@ -21,6 +21,8 @@
 use TYPO3\CMS\Backend\Controller\LoginController;
 use TYPO3\CMS\Backend\LoginProvider\LoginProviderInterface;
 use TYPO3\CMS\Core\Page\PageRenderer;
+use TYPO3\CMS\Core\TypoScript\TypoScriptService;
+use TYPO3\CMS\Extbase\Configuration\ConfigurationManager;
 use TYPO3\CMS\Fluid\View\StandaloneView;
 use Waldhacker\Oauth2Client\Service\Oauth2ProviderManager;
 
@@ -30,14 +32,28 @@
 
     private Oauth2ProviderManager $oauth2ProviderManager;
 
-    public function __construct(Oauth2ProviderManager $oauth2ProviderManager)
-    {
+    protected ConfigurationManager $configurationManager;
+
+    protected TypoScriptService $typoScriptService;
+
+    public function __construct(
+        Oauth2ProviderManager $oauth2ProviderManager,
+        ConfigurationManager $configurationManager,
+        TypoScriptService $typoScriptService
+    ) {
         $this->oauth2ProviderManager = $oauth2ProviderManager;
+        $this->configurationManager = $configurationManager;
+        $this->typoScriptService = $typoScriptService;
     }
 
     public function render(StandaloneView $view, PageRenderer $pageRenderer, LoginController $loginController)
     {
-        $view->setTemplatePathAndFilename('EXT:oauth2_client/Resources/Private/Templates/Backend/Oauth2LoginProvider.html');
+        $configuration = $this->configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_FULL_TYPOSCRIPT);
+        $typoScript = $this->typoScriptService->convertTypoScriptArrayToPlainArray($configuration);
+
+        $view->setTemplateRootPaths($typoScript['plugin']['tx_oauth2client']['view']['templateRootPaths']);
+        $view->setTemplate('Backend/Oauth2LoginProvider');
+
         $view->assign('providers', $this->oauth2ProviderManager->getConfiguredBackendProviders());
     }
 }
